packages/docs/developer-resources/walkthroughs/web-dapp.md

```diff
@@ -0,0 +1,299 @@
+# Building a simple DApp with React
+
+This tutorial will be a basic guide on developing a decentralised application (DApp) on top of Celo. We'll be developing against one of the core Celo contracts, [Governance.sol](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/governance/Governance.sol), and allowing users of our DApp to vote on active [Celo Governance proposals](https://docs.celo.org/celo-owner-guide/voting-governance).
+
+## Foreword
+
+This guide requires an understanding of a few popular web technologies. Our application will be written in [React](https://reactjs.org/), utilising [hooks](https://reactjs.org/docs/hooks-intro.html) for state management and built with [Next.js](https://nextjs.org/), a popular static site generation framework.
+
+If you find this tutorial lacking in any way or want to dive into the code more thoroughly, checkout the [Celo Tools](https://github.com/alexbharley/celo-tools) GitHub repository where much of this tutorial has been ported from.
+
+## Getting started
+
+Step one of developing our application is scaffolding it out with `create-next-app` and adding TypeScript compilation so we can develop more confidently.
+
+```bash
+yarn create next-app voting-dapp
+cd voting-dapp
+touch tsconfig.json
+yarn add --dev typescript @types/react @types/node
+```
+
+Now running `yarn dev` should open up our new Next.js website on `localhost:3000`.
+
+Next we'll need to add a few Celo specific dependencies so we can work with our core contracts.
+
+```bash
+yarn add @celo/contractkit @celo-tools/use-contractkit bignumber.js
+```
+
+Here's what we'll be using each of these packages for:
+
+- [@celo/contractkit](https://github.com/celo-org/celo-monorepo/tree/master/packages/sdk/contractkit) is a lightweight wrapper around the [Web3](https://web3js.readthedocs.io/) object you may alrady be familiar with. It contains typed interfaces for the core contracts (generated from the Contract ABIs) and helper functions to make common operations on Celo easier
+- [@celo-tools/use-contractkit](https://github.com/celo-tools/use-contractkit) is a community provided library to ease establishing the connection with a users wallet. When developing with this library, your users can hold Celo via Valora, a Ledger, Metamask and more
```

Comment:
Suggest adding: 
"...establishing the connection with a users wallet, whether that is a hardware wallet, mobile wallet, or web wallet."

"...[Valora](https://valoraapp.com/), a Ledger, Metamask and more"